pkl-cli/pkl-cli.gradle.kts

```
@@ -152,12 +158,50 @@ val testStartJavaExecutable by
     doLast { outputFile.get().asFile.writeText("OK") }
   }
 
-tasks.check { dependsOn(testStartJavaExecutable) }
+// Setup `testStartJavaExecutable` tasks for multi-JDK testing.
+val testStartJavaExecutableOnOtherJdks =
+  buildInfo.jdkTestRange.map { jdkTarget ->
+    val taskName = "testStartJavaExecutableJdk${jdkTarget.asInt()}"
+    val toolchains = project.serviceOf<JavaToolchainService>()
+
+    tasks.register(taskName, Exec::class) {
+      // dummy output to satisfy up-to-date check
+      val outputFile = layout.buildDirectory.file(taskName)
+      outputs.file(outputFile)
+      dependsOn(javaExecutable)
+      executable =
+        toolchains
+          .launcherFor { languageVersion = jdkTarget }
+          .get()
+          .executablePath
+          .asFile
+          .absolutePath
```

nit: don't call `get()` on a provider during configuration phase

```suggestion
      setExecutable(
        toolchains
          .launcherFor { languageVersion = jdkTarget }
          .map { it.executablePath.asFile.absolutePath }
      )
```

---

this didn't work; turns out the `setExecutable` method will; accept that, but it does not know how to use it. in any case, this task is created via `register`, so it should be configured lazily anyway